platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -343,10 +350,15 @@ public class SwirldsPlatform implements Platform, Startable {
 
         this.eventMapper = new EventMapper(platformContext.getMetrics(), selfId);
 
+        platformStatusStateMachine = components.add(buildPlatformStatusStateMachine(time));
+
         this.metrics = platformContext.getMetrics();
 
         metrics.getOrCreate(StatConstructor.createEnumStat(
-                "PlatformStatus", Metrics.PLATFORM_CATEGORY, PlatformStatus.values(), currentPlatformStatus::get));
+                "PlatformStatus",
+                Metrics.PLATFORM_CATEGORY,
+                PlatformStatus.values(),
+                platformStatusStateMachine::getCurrentStatus));
```

Comment:
Can this be encapsulated within the state machine class?

Reply:
Unfortunately, doing so creates a circular dependency. I'm assuming the `common` module isn't intended to ever include something from `core`?

Reply:
Hmm... let's not try and solve this in this PR. Your changes didn't cause this problem.

As I slowly make additional changes to clean up `SwirldsPlatform.java`, I will probably just replace this "enum stat" with a basic lambda. Not something you need to do for this PR though.